core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/StringTerms.java

```diff
@@ -67,6 +67,12 @@ public Object getKey() {
         @Override
         public Number getKeyAsNumber() {
             // this method is needed for scripted numeric aggs
+            try {
+                return Long.parseLong(termBytes.utf8ToString());
+            } catch (NumberFormatException ignored) {
+            }
+            //first the number is tried to be parsed as a Long
+            //if it fails, it will be parsed as a Double
```

Comment:
This comment is a translation of the code to English, information that can already be gleaned by reading the code. The comment I am looking for here is *why* the code is written the way that it is written. *Why* are we doing this, why is it okay to swallow an exception?